### PR TITLE
Navbar

### PR DESCRIPTION
**Pull request by geraxrodriguez**
Added styled navbar

Added two more folders for routing. 4 total now (about, contact, placeholder, and login). 

Added LPDD logo.

Tested on mobile and desktop screens. 

Known issues:
- [ ] if more navlinks are added, it would look crammed on mobile. Ideally, we update the design to implement a hamburger menu

### CODE REVIEW

File: src/app/ui/nav-bar.tsx [hunk 1]
```diff
@@ -1,13 +1,34 @@
...
+              alt="Logo for Latiné Professional Development Directory"
+            />
+            <p className="text-xl text-white tracking-tighter">LPDD</p>
+        </Link>
+        <NavLinks />
+      </div>
+    </nav>
   );
 }
+
```
Comment (jvmontes) [inline src/app/ui/nav-bar.tsx:34]: I took a look at this. I see how you were using flex-col to align the items center. I believe I came up with an alternate solution that accomplishes the same objective but cleans up the code a bit. 

Change would be here:

```
  const navClasses = clsx(
    "w-full h-14 flex items-center justify-between px-2 bg-slate-900",
    pathname === "/" && "absolute z-10 bg-transparent"
  );
```

So we combine `flex items-center justify-between` and then just remove a layer (that first div inside the nav)

```
  return (
    <nav className={navClasses}>
      <Link href="/" className="flex items-center mr-12">
        <Image...
```

Let me know what you think about this approach... copied the full component below with changes for additional clarity. 

```
"use client";

import Link from "next/link";
import NavLinks from "./nav-links";
import { usePathname } from "next/navigation";
import clsx from "clsx";
import Image from "next/image";

export default function NavBar() {
  const pathname = usePathname();

  const navClasses = clsx(
    "w-full h-14 flex items-center justify-between px-2 bg-slate-900",
    pathname === "/" && "absolute z-10 bg-transparent"
  );

  return (
    <nav className={navClasses}>
      <Link href="/" className="flex items-center mr-12">
        <Image
          src="/logo.svg"
          width={40}
          height={40}
          alt="Logo for Latiné Professional Development Directory"
        />
        <p className="text-xl text-white tracking-tighter">LPDD</p>
      </Link>
      <NavLinks />
    </nav>
  );
}
```

File: src/app/ui/nav-links.tsx [hunk 2]
```diff
@@ -6,31 +6,45 @@ import clsx from "clsx";
 
 const links = [
   {
-    name: "About",
+    name: "ABOUT",
```
Comment (jvmontes) [inline src/app/ui/nav-links.tsx:9]: be ABOUT it

File: src/app/ui/nav-links.tsx [hunk 3]
```diff
@@ -6,31 +6,45 @@ import clsx from "clsx";
...
     href: "/about",
   },
   {
-    name: "Contact",
+    name: "CONTACT",
```
Comment (jvmontes) [inline src/app/ui/nav-links.tsx:13]: if this change was done for styling purposes, we also could `.toUppercase()` the strings we display to ensure they're showing up as expected.

File: src/app/ui/nav-links.tsx [hunk 4]
```diff
@@ -6,31 +6,45 @@ import clsx from "clsx";
...
     href: "/contact",
   },
+  {
+    name: "PLACEHOLDER",
+    href: "/placeholder",
```
Comment (jvmontes) [inline src/app/ui/nav-links.tsx:18]: please remove placeholders

File: src/app/ui/nav-links.tsx [hunk 5]
```diff
@@ -6,31 +6,45 @@ import clsx from "clsx";
...
+  },
+  {
+    name: "LOGIN",
+    href: "/login",
+  },
```
Comment (jvmontes) [inline src/app/ui/nav-links.tsx:23]: let's comment out this object for now until the login functionality is closer to being built out.

File: src/app/ui/nav-links.tsx [hunk 6]
```diff
@@ -6,31 +6,45 @@ import clsx from "clsx";
...
 ];
 
+
 export default function NavLinks() {
   const pathname = usePathname();
+  
+  const baseClasses = "text-[0.70rem] pr-2 tracking-tighter sm:text-sm sm:pl-8 sm:tracking-normal";
```
Comment (jvmontes) [inline src/app/ui/nav-links.tsx:30]: should we remove the `sm` qualifiers and consider running past the designers to find a good `md` option?

File: src/app/ui/nav-links.tsx [hunk 7]
```diff
@@ -6,31 +6,45 @@ import clsx from "clsx";
...
 
         return (
-          <Link key={link.name} href={link.href}>
+          <Link key={name} href={href} className="flex items-center">
             <p
               className={clsx(
-                "px-4",
-                isActiveLink ? "text-gray-800 underline" : "text-blue-800"
+                baseClasses,
+                isActiveLink ? activeClasses : inactiveClasses
```
Comment (jvmontes) [inline src/app/ui/nav-links.tsx:44]: great clean up. well done.

File: src/app/ui/nav-links.tsx [hunk 8]
```diff
@@ -6,31 +6,45 @@ import clsx from "clsx";
...
-    <div className="flex justify-end">
-      {links.map((link) => {
-        const isActiveLink = pathname === link.href;
+    <div className="flex">
+      {links.map(({ name, href}) => {
+        const isActiveLink = pathname === href;
 
         return (
-          <Link key={link.name} href={link.href}>
+          <Link key={name} href={href} className="flex items-center">
```
Comment (jvmontes) [inline src/app/ui/nav-links.tsx:40]: I just tested removing `className="flex items-center"` and it looks like it caused no change. Can you test this on your end and if it's not necessary, can we remove it?

File: src/app/ui/home-hero.tsx [hunk 9]
```diff
@@ -15,7 +15,7 @@ const lexendMedium = Lexend({
 export default function HomeHero() {
   return (
     <section className="relative w-full h-[50vh] md:h-[75vh] bg-black"> 
-      <div className="w-full h-full"> 
+      <div className="w-full h-full mr-1">
```
[the remark this answers is not in the document]
Comment (geraxrodriguez) [reply src/app/ui/home-hero.tsx:18]: hmm, not sure. Deleted it. Apologies.

File: src/app/ui/nav-bar.tsx [hunk 10]
```diff
@@ -1,13 +1,34 @@
...
+import { usePathname } from 'next/navigation';
+import clsx from 'clsx';
+import Image from "next/image";
 
 export default function NavBar() {
+  const pathname = usePathname();
+
+  const navClasses = clsx(
+    "w-full h-14 flex flex-col justify-center bg-slate-900",
+    pathname === '/' && "sm:absolute sm:z-10 sm:bg-transparent"
```
[the remark this answers is not in the document]
Comment (geraxrodriguez) [reply src/app/ui/nav-bar.tsx:14]: good catch. the first two have been added for all sizes. 

`sm:bg-transparent` needs to stay as is so that the background color shows when on the home page for mobile.

File: src/app/ui/nav-links.tsx (hunk 4 above)
Comment (geraxrodriguez) [reply src/app/ui/nav-links.tsx:18]: Removed.

File: src/app/ui/nav-links.tsx (hunk 8 above)
Comment (geraxrodriguez) [reply src/app/ui/nav-links.tsx:40]: No change!! Removed.

File: src/app/ui/nav-links.tsx (hunk 6 above)
Comment (geraxrodriguez) [reply src/app/ui/nav-links.tsx:30]: We can. I'll resolve this for now, but will make note of it that we'll be discussing this with designers.

File: src/app/placeholder/page.tsx [hunk 11]
```diff
@@ -0,0 +1,7 @@
+export default function Page() {
```
Comment (jvmontes) [inline src/app/placeholder/page.tsx:1]: delete this?

File: src/app/ui/nav-bar.tsx [hunk 12]
```diff
@@ -1,13 +1,47 @@
...
 import NavLinks from "./nav-links";
+import { usePathname } from 'next/navigation';
+import clsx from 'clsx';
+import Image from "next/image";
 
 export default function NavBar() {
+  const pathname = usePathname();
+
+  const navClasses = clsx(
+    // "w-full h-14 flex flex-col justify-center bg-slate-900 absolute z-10",
```
Comment (jvmontes) [inline src/app/ui/nav-bar.tsx:13]: please delete and do not commit commented out code

File: src/app/ui/nav-bar.tsx [hunk 13]
```diff
@@ -1,13 +1,47 @@
...
+    //         src="/logo.svg"
+    //         width={40}
+    //         height={40}
+    //         alt="Logo for Latiné Professional Development Directory"
+    //       />
+    //       <p className="text-xl text-white tracking-tighter">LPDD</p>
+    //     </Link>
+    //     <NavLinks />
+    //   </div>
+    // </nav>
```
Comment (jvmontes) [inline src/app/ui/nav-bar.tsx:32]: please delete commented code

File: src/app/ui/nav-links.tsx [hunk 14]
```diff
@@ -4,33 +4,48 @@ import Link from "next/link";
 import { usePathname } from "next/navigation";
 import clsx from "clsx";
 
+// home - about - contact - login
```
Comment (jvmontes) [inline src/app/ui/nav-links.tsx:7]: idk if this comment is necessary. ask chat gpt about Clean Code by Uncle Bob & self-documenting code. 

Spoiler alert: answer below


> Ah, yes! Uncle Bob emphasizes the importance of clean code and the idea that code should be self-documenting. This means that well-written code, by itself, should be understandable without needing excessive comments. Here's a quick refresher on the key principles:
> 
> Meaningful Names: Variables, functions, and classes should have names that reveal their intent. This makes the code easier to understand just by reading it.
> 
> Single Responsibility Principle (SRP): Each function or class should do one thing, and do it well. This avoids complex, multi-purpose blocks of code that are hard to follow or modify.
> 
> Avoid Comments: Comments are often redundant if your code is clean and self-explanatory. Instead of commenting on what the code does, write the code clearly enough that the intent is obvious.
> 
> Small Functions: Functions should be short and focused. A good rule is that they should fit on one screen without scrolling.
> 
> Consistent Formatting: Maintain consistent indentation, spacing, and line breaks. This makes the code easier to visually parse and contributes to readability.
> 
> DRY (Don't Repeat Yourself): Avoid duplicating code. If you find the same logic repeated in multiple places, refactor it into a reusable function or class.
> 
> Error Handling: Handle errors in a way that makes the flow of the code still easy to understand.
> 
> These ideas contribute to self-documenting code, where the code is its own explanation. It's readable and makes sense to any developer picking it up, like a well-written book!

File: src/app/ui/nav-links.tsx [hunk 15]
```diff
@@ -4,33 +4,48 @@ import Link from "next/link";
...
   },
   {
-    name: "Contact",
+    name: "CONTACT",
     href: "/contact",
   },
+  // {
+  //   name: "LOGIN",
+  //   href: "/login",
+  // },
```
Comment (jvmontes) [inline src/app/ui/nav-links.tsx:24]: im ok with leaving this comment here since it'll hopefully be tackled soon

File: src/app/ui/nav-links.tsx [hunk 16]
```diff
@@ -4,33 +4,48 @@ import Link from "next/link";
...
 ];
 
+
 export default function NavLinks() {
   const pathname = usePathname();
+  
+  const baseClasses = "text-[0.70rem] pr-2 tracking-tighter sm:text-sm sm:pl-8 sm:tracking-normal";
```
Comment (jvmontes) [inline src/app/ui/nav-links.tsx:31]: preference to use something like text-base, text-sm, text-md, text-lg, etc. over explicit rem values. Please try to find a pre-existing tailwind class that can work, if possible, prefferably.

**Review by jvmontes — CHANGES_REQUESTED** [review]

File: src/app/ui/nav-links.tsx (hunk 16 above)
Comment (jvmontes) [reply src/app/ui/nav-links.tsx:31]: I am removing the basClasses const because if these Tailwind Classes aren't in a className attribute, when you hover over them, it doesn't show the actual CSS class tooltip provided by the Tailwind CSS IntelliSense plugin.

**Review by jvmontes — APPROVED** [review]
